### DISCUSSION TITLE
In nuxt3, how does pinia get a store by ID

### DISCUSSION THREAD

**Discussion by gllinzbk1** [vuejs/pinia#2129] [Help and Questions] [answered]
Describe the feature
In nuxt3, how does pinia get a store by ID，This function can be implemented？
```
export const registerStore = (category_id) => {
  return category_id = defineStore(category_id, {
    state: () => ({
    products: [],
    allTags: [],
    }),
    getters: {},
    actions: {}
  })
}
```
//-------------------------use layer1
```
const productStore = registerStore(category_id)();
productStore.products =['a','b','c'....]
productStore.allTags =['a','b','c'....]
```

//----------------------get,maybe,i have many layers,just first layer set data,others layer just get store by id
how to get a store by ID?
like: ```getstore(category_id)```

The purpose is to store different categories of data in different stores

**Comment by ilyarefraction** [comment, marked as answer]
@gllinzbk1 See the last two commits in this repo: https://github.com/maIIady/pinia-tests

Possible solution 1: make a simple function that accesses the internal `_s` property. Smth like
```js
const getStoreById = (id: string) => {
  //@ts-expect-error _s is internal
  return getActivePinia()?._s.get(id);
};
```
This should work, but I don't think there is a way to get proper typing for this implementation (see mentioned repo)

Solution 2:
Use pinia plugin API to place registered stores in your store.
```js
const pinia = createPinia();
pinia.use(({ store }) => {
  const globalStore = useGlobalStore();

  if (!globalStore.allPiniaStores.has(store.$id)) {
    globalStore.allPiniaStores.set(store.$id, store);
  }

  const { $dispose } = store;
  store.$dispose = () => {
    $dispose();
    globalStore.allPiniaStores.delete(store.$id);
  };
});
```

 This is a better solution because you do not rely on the pinia's internals, however, typing is problematic here too. You can see the last commit in the repo to check the actual code.

**Reply by gllinzbk1** [reply]
@maIIady getActivePinia()?._s.get(id);
Is there a limit on the number of acquisitions, only acquisitions are available in ‘NPM run generation’
